fraim/workflows/infrastructure_discovery/workflow.py

```diff
@@ -0,0 +1,331 @@
+# SPDX-License-Identifier: MIT
+# Copyright (c) 2025 Resourcely Inc.
+
+"""
+Infrastructure Discovery Workflow
+
+Analyzes infrastructure and deployment configurations to identify container configurations,
+orchestration patterns, scaling policies, and deployment strategies.
+"""
+
+import os
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Annotated, Any, Dict, List, Optional
+
+from pydantic import BaseModel
+
+from fraim.config import Config
+from fraim.core.contextuals import CodeChunk
+from fraim.core.llms.litellm import LiteLLM
+from fraim.core.parsers import PydanticOutputParser
+from fraim.core.prompts.template import PromptTemplate
+from fraim.core.steps.llm import LLMStep
+from fraim.core.workflows import ChunkProcessingMixin, ChunkWorkflowInput, Workflow
+from fraim.workflows.registry import workflow
+from fraim.workflows.utils import write_json_output
+
+# Infrastructure-focused file patterns
+INFRASTRUCTURE_FILE_PATTERNS = [
+    # Infrastructure & Container files
+    "Dockerfile",
+    ".dockerfile",
+    "docker-compose.yml",
+    "docker-compose.yaml",
+    "*.k8s.yaml",
+    "*.k8s.yml",
+    "deployment.yaml",
+    "service.yaml",
+    "ingress.yaml",
+    "*.tf",
+    "*.tfvars",
+    "*.hcl",
+    "terraform.tfstate",
+    # Infrastructure Configuration files
+    "*.yaml",
+    "*.yml",
+    "*.json",
+    "*.toml",
+    "*.ini",
+    "*.conf",
+    "*.config",
+    "*.properties",
+    "*.env",
+    ".env*",
+    "*.settings",
+    # Build & Package files (reveal deployment structure)
+    "package.json",
+    "requirements.txt",
+    "Pipfile",
+    "pyproject.toml",
+    "pom.xml",
+    "build.gradle",
+    "Cargo.toml",
+    "composer.json",
+    "Gemfile",
+    "Makefile",
+    "makefile",
+    "*.mk",
+    # Orchestration and CI/CD files
+    ".github/workflows/*.yml",
+    ".github/workflows/*.yaml",
+    ".gitlab-ci.yml",
+    "azure-pipelines.yml",
+    "Jenkinsfile",
+    "skaffold.yaml",
+    "helm/**/*.yaml",
+    "helm/**/*.yml",
+]
+
+# Load infrastructure prompts
+INFRASTRUCTURE_PROMPTS = PromptTemplate.from_yaml(
+    os.path.join(os.path.dirname(__file__), "infrastructure_prompts.yaml")
+)
+
+
+# Pydantic models for Infrastructure Analysis
+class EnvironmentVariable(BaseModel):
+    name: str
```

Comment:
We should include a description for each field in these pydantic models. When PydanticOutputParser generates the output schema to inject into the prompt, those descriptions will be included.